docs/source/specification.md

```diff
@@ -0,0 +1,179 @@
+
+There might be good reasons why any particular TRE does not possess one or more of the capabilities listed in this specification, but most TREs should aspire to meet them in the long-term.
+
+(tre_roles)=
+
+## Roles
+
+A TRE conforming to the SATRE specification should provide a broadly similar experience for stakeholders operating in each of these defined roles.
+There is not necessarily a one-to-one mapping between roles and people.
+One person can have multiple roles.
+
+### TRE users
+
+The researchers working on projects that involve logging into a TRE to access data.
+
+<!-- The document will explain that user experience of the platform and associated documentation should feel similar across TREs conforming to SATRE specification. -->
+
+```{list-table}
+:header-rows: 1
+:name: tab-tre-role-user
+
+* - Role name
+  - Role description
+* -
+  -
+```
+
+### TRE administration roles
+
+The IT and related professionals who will be responsible for deploying and managing instances of a TRE conforming to the SATRE specification.
+These roles cover managing TRE computing infrastructure, but also administering the TRE itself (_e.g._ managing users and projects).
+
+<!-- The document will explain that SATRE conforming TREs should have documentation and infrastructure deployment code/apps that conform to software engineering best practices, which are also defined here, making them "simple" for an IT professional to follow; troubleshooting steps included. -->
+
+```{list-table}
+:header-rows: 1
+:name: tab-tre-role-administrator
+
+* - Role name
+  - Role description
+* -
+  -
+```
+
+### TRE developer roles
+
+The software engineers responsible for developing and maintaining TRE software, including adding functionality, bug fixes and general maintenance.
+
+<!-- The document will explain recommended practices suitable for developing a software of this complexity and reference learnings from existing TRE developers. -->
+
+```{list-table}
+:header-rows: 1
+:name: tab-tre-role-developer
+
+* - Role name
+  - Role description
+* -
+  -
+```
+
+### TRE governance roles
+
+Roles that uphold the governance of TREs.
+Such governance responsibilities typically involve establishing policies and procedures to ensure the responsible use of data, protecting the privacy and confidentiality of research participants, and promoting transparency and accountability in research activities.
+Typical roles might include data custodians, ethicists, an independent board or a lay panel.
+
+```{list-table}
+:header-rows: 1
+:name: tab-tre-role-governance
+
+* - Role name
+  - Role description
+* -
+  -
+```
+
+## Specification
+
+Details of the SATRE specification are shown below, together with the breakdown of the pillars into capabilities and components.
+
+```{toctree}
+:numbered:
+pillars/information_governance.md
+pillars/computing_technology.md
+pillars/data_management.md
+pillars/supporting.md
+```
```

Comment:
Is there a way to also add this to the 'main' toctree?

When I click on 'Specification' in the sidebar it doesn't expand to show each pillar and each pillar's capabilities. That only appears when I navigate to one of the pillars/capabilities.

Reply:
I couldn't work out how to do this while using the RTD theme. I think nesting-depth of the sidebar is a theme-specific option. If you find a configurable setting I'll give this a go.

Reply:
No, `navigation depth` doesn't help. MyST builds the left-hand bar based *either* on headings *or* on the toctree. However, the toctree can only include links to other pages, not links inside the document. There's therefore no way to combine links to other docs and internal links in the same sidebar.

Reply:
That doesn't seem to work locally.

Maybe this is something @manics can help with. I've never understood sphinx TOC.

Reply:
Ah, I see. It actually isn't using the `toctree` directives because we can see subsections of pages?
I'm not sure _why_ it isn't using the directives as those seem correct :confused:.

I would think it is better to use the directives.
Does the theme allow us to have a separate navigation element for sections within a page like the [myst-parser docs](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html)?

Reply:
Changing to a different theme _e.g._ [book](https://sphinx-themes.readthedocs.io/en/latest/sample-sites/sphinx-book-theme/) fixes this. Should we just move to that?

It gives us better control of the ToC and structure.

Reply:
Can we make that change in a separate PR, and merge this?

Reply:
If we want to stick with the rtd theme we can have a layout like this by putting multiple toctree directives on the index page.

<img width="276" alt="Screenshot 2023-06-22 at 11 15 59" src="https://github.com/sa-tre/satre-specification/assets/3502751/677b6a52-8193-4aff-b157-bf719e436f78">

What do you think?

Reply:
To be honest, there's actually so little useful content on `index.md` now that I wonder whether the answer is to use this structure and move the remaining overview/introductory content in `specification.md` to `index.md`?

Reply:
That looks good (also, helps get this change merged in, we can worry about theming later).

I also think the specification section was getting too long with the introductory material. Might be worth breaking that down too (not necessarily here).